Test-№1.2-from-09.11.2022/task1.c

```diff
@@ -0,0 +1,123 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <math.h>
+
+void swap(int *firstElement, int *secondElement)
+{
+    int temp = *firstElement;
+    *firstElement = *secondElement;
+    *secondElement = temp;
+}
+
+void printArray(int *array, int lenOfArray)
+{
+    for (int i = 0; i < lenOfArray; i++)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+
+void evenOddSort(int *array, int lenOfArray)
+{
+    int leftPointer = 0, rightPointer = lenOfArray - 1;
+    while (leftPointer < rightPointer)
+    {
+        if (array[leftPointer] % 2 == 1)
+        {
+            if (array[rightPointer] % 2 == 0)
+            {
+                swap(&array[leftPointer], &array[rightPointer]);
+                leftPointer++;
+                rightPointer--;
+            }
+            else
+            {
+                rightPointer--;
+            }
+        }
+        else
+        {
+            leftPointer++;
+        }
+    }
+}
+
+bool isCorrArray(int *array, int lenOfArray)
+{
+    bool answer = true;
+    bool AreNotWeChangedTheNowOst = true;

```

Comment:
```suggestion
    bool stillEven = true;

```
Я так и не понял, что значит это название переменной

Reply:
И эта переменная не нужна, ведь переменной с остатком вполне достаточно.